src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_data_plane_commands.py

```diff
@@ -376,22 +356,79 @@ def test_batch_task_create_cmd(self, resource_group, batch_account_name):
         self.assertEqual(len(result), 3)
         self.assertTrue(any([i for i in result if i['taskId'] == 'xplatTask1']))
 
+
+    @ResourceGroupPreparer()
+    @BatchAccountPreparer(location='eastus2')
+    def test_batch_file_download_cmd(self, resource_group, batch_account_name):
+        self.set_account_info(batch_account_name, resource_group)
+        self.kwargs.update({
+            'j_id': 'xplatJobForTaskTests',
+            't_id': 'xplatTask1',
+            'p_id': 'xplatTestPool',
+            'j_file': self._get_test_data_file('batchCreateJobForTaskTests.json'),
+            'p_file': self._get_test_data_file('batchCreateTestPool.json'),
+            'ts_file': self._get_test_data_file('batchCreateMultiTasks.json')
+        })
+        # create a pool, job and task
+        self.batch_cmd('batch pool create --json-file "{p_file}"')
+        self.batch_cmd('batch job create --json-file "{j_file}"')
+        self.batch_cmd('batch task create --job-id {j_id} --json-file "{ts_file}"')
+
+        self.wait_for_task_complete("xplatJobForTaskTests",3)
+
+        # verify task file download
+        if os.path.isfile('c:/temp/stdout.txt'): os.remove('c:/temp/stdout.txt')
+        self.batch_cmd('batch task file download --destination c:/temp/stdout.txt --file-path stdout.txt --job-id {j_id} --task-id {t_id}')
+        self.assertTrue(os.path.isfile('c:/temp/stdout.txt'), 'File c:/temp/stdout.txt does not exist.')
+        self.assertGreater(os.path.getsize('c:/temp/stdout.txt'), 0, 'File c:/temp/stdout.txt is empty.')
```

Comment:
Please avoid use `c:/temp/` as PR pipeline runs in linux. Try `tempfile.TemporaryDirectory` to fix CI failure

Reply:
This should be resolved now